Sources/PackageGraph/RepositoryPackageContainerProvider.swift

```diff
@@ -161,15 +161,23 @@ public class RepositoryPackageContainer: PackageContainer, CustomStringConvertib
     }
 
     public func getDependencies(at version: Version) throws -> [RepositoryPackageConstraint] {
+        // FIXME: We should have a persistent cache for these.
+        let tag = knownVersions[version]!
+        let revision = try repository.resolveRevision(tag: tag)
+        return try getDependencies(at: revision, version: version)
+    }
+
+    public func getDependencies(at revision: String) throws -> [RepositoryPackageConstraint] {
+        return try getDependencies(at: Revision(identifier: revision))
+    }
+
+    /// Returns dependencies of a container at the given revision.
+    private func getDependencies(at revision: Revision, version: Version? = nil) throws -> [RepositoryPackageConstraint] {
```

Comment:
👍 -- FWIW I would probably land something like this in its own patch just to separate obvious infra improvement from hard algorithm things